lua/lspconfig/server_configurations/clangd.lua

```diff
@@ -2,10 +2,15 @@ local util = require 'lspconfig.util'
 
 -- https://clangd.llvm.org/extensions.html#switch-between-sourceheader
 local function switch_source_header(bufnr)
+  local clangd_client = nil
+  for _, v in ipairs(vim.lsp.buf_get_clients(0)) do
+    if v.name == 'clangd' then
+      clangd_client = v
+    end
+  end
   bufnr = util.validate_bufnr(bufnr)
   local params = { uri = vim.uri_from_bufnr(bufnr) }
-  vim.lsp.buf_request(
-    bufnr,
+  clangd_client.request(
```

Comment:
what happens if `clangd_client` is `nil`?

Reply:
Sure I could add a conditional `if clang_client` if you wish, but seeing as the the command is local to the language server, definied later in the document with `commands = {`, would it cause the clang_client to be nil?

Reply:
The server could have crashed, or the user could have switched to a different buffer with a different filetype (which doesn't "undefine" the command!).

Reply:
The important thing is that it handles this case gracefully and doesn't throw a Lua error.

Reply:
Good point. will add the conditional and the previous suggestions! Thanks